grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -252,7 +259,13 @@ public void start(Listener<O> responseListener, Metadata metadata) {
                 deframed.subscribe(this, ctx.eventLoop(), SubscriptionOption.WITH_POOLED_OBJECTS);
             });
         }
-        responseListener.onReady();
+        contextAwareExecutor.execute(() -> {
+            try {
+                responseListener.onReady();
+            } catch (Throwable t) {
+                closeWhenListenerThrows(t);
+            }
+        });
```

Comment:
What do you think of adding a wrapping `Listener` that schedules for invoking the methods in `Listener` if `callOptions.getExecutor()` is specified? The current style can reduce duplicate code but allocate additional objects to create lambda.

```java
void start(...) {
   if (callOptions.getExecutor() == null) {
        listener = responseListener;
   } else (
       listener = new CallExecutorAwareListener(responseListener, callOptions.getExecutor());
   }
}

class CallExecutorAwareListener<T> extends Listener<T> {

    private final Listener<T> delegate;
    private final Executor executor;

    CallExecutorListener(Listener<T> delegate, Executor executor) {
        this.delegate = delegate;
        this.executor = executor;
    }

    @Override
    public void onHeaders(Metadata headers) {
        executor.execute(() -> {
            try {
                delegate.onHeaders(headers);
            } catch (Throwable t) {
                closeWhenListenerThrows(t);
            }
        });
    }
    ...
}
```

Reply:
Updated the PR. PTAL!